### DISCUSSION TITLE
Why GPL3?

### DISCUSSION THREAD

**Discussion by Lemiort** [adepierre/Botcraft#51] [Q&A] [answered]
Why have you chosen GPL3 license? It's quite strict. 
E.g., it permits linking the library to the most of the graphics engines.

**Comment by adepierre** [comment, marked as answer]
The main idea is that I don't want people to build bots using botcraft to grief servers without open-sourcing the code, so server owners can more easily counter them.

Now I'm not an expert of license, but with a more permissive license like MIT, I think it would be allowed to do such things without sharing the code, while it is not under GPL3.

**Reply by Dashbloxx** [reply]
Compiling botcraft as a shared library would not require the person to share their code because it isn't linking with GPLv3 code.

**Reply by GEOEGII555** [reply]
Sharing the code is only required if you're distributing the program. You don't have to share the code if you keep the program private.

**Reply by BlueCannonBall** [reply]
If you want to require them to share the code even if they aren't distributing binaries of their bot, use the AGPL. That way, communicating with the bot online would count as distribution.
